src/App.tsx

```diff
@@ -1,28 +1,100 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Sort } from './types/Sort';
+import { filterTodos } from './services/filterTodos';
 
-const USER_ID = 0;
+
+
+const USER_ID = 11932;
+
+type ErrorMessageType = React.Dispatch<React.SetStateAction<string | null>>;
+type MessageType = string;
+
+const handleErrorMessage = (
+  setErrorMessage: ErrorMessageType,
+  message: MessageType,
+) => {
+  setErrorMessage(message);
+  setTimeout(() => {
+    setErrorMessage(null);
+  }, 3000);
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState(Sort.All);
+  const [isEditing, setIsEditing] = useState<Record<number, boolean>>({});
+
+
+  const visibleTodos = useMemo(() => {
+    return filterTodos(todos, selectedFilter);
+  }, [todos, selectedFilter]);
+
+
+  const handleToggleEditing = (id: number) => {
+    setIsEditing((prevIsEditing) => ({
+      ...prevIsEditing,
+      [id]: !prevIsEditing[id],
+    }));
+  };
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const fetchedTodos = await getTodos(USER_ID);
+
+        setTodos(fetchedTodos);
+      } catch (error) {
+        if (error instanceof Error) {
+          handleErrorMessage(setErrorMessage,
+            error.message || 'Unable to load todos');
+        } else {
+          handleErrorMessage(setErrorMessage, 'Unable to load todos');
+        }
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const isActiveTodo = todos.every(todo => !todo.completed);
+  const isNotCompleted = todos.filter(todo => !todo.completed).length;
```

Comment:
we usually name boolean vars using the "is", it does not look like the boolean var
```suggestion
  const nrOfNotCompleted = todos.filter(todo => !todo.completed).length;
```